### PR TITLE
Enhanced RDoc for Enumerable

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Treated:
- #none?
- #one?
- #min

### CODE REVIEW

File: enum.c [hunk 1]
```diff
@@ -2026,26 +2063,52 @@ min_ii(RB_BLOCK_CALL_FUNC_ARGLIST(i, args))
...
- *  first form assumes all objects implement <code><=></code>;
- *  the second uses the block to return <em>a <=> b</em>.
+ *  Returns the element with the minimum value according to a given criterion.
+ *  The ordering of equal elements is indeterminate and may be unstable.
  *
- *     a = %w(albatross dog horse)
- *     a.min                                   #=> "albatross"
- *     a.min { |a, b| a.length <=> b.length }  #=> "dog"
+ *  With no argument and no block, returns the minimum value,
+ *  using the elements' own method <tt>=></tt> for comparison:
```
Comment (jeremyevans) [inline enum.c:2075]: `=>` => `<=>`

**Review by jeremyevans — CHANGES_REQUESTED** [review]
Looks good!  Only one minor change requested, OK to merge after that.

File: enum.c (hunk 1 above)
Comment (BurdetteLamar) [reply enum.c]: Done.

File: enum.c [hunk 2]
```diff
@@ -1904,26 +1904,45 @@ rb_nmin_run(VALUE obj, VALUE num, int by, int rev, int ary)
...
+ *    {foo: 0}.one?(Array)                 # => true
+ *    [].one?(Integer)                     # => false
+ *
+ *  With a block given, returns whether the block returns a truthy value
+ *  for exactly one element:
+ *
+ *    (1..4).one? {|element| element < 2 }                     # => true
+ *    (1..4).one? {|element| element < 1 }                     # => false
+ *    {foo: 0, bar: 1, baz: 2}.one? {|key, value| value < 1 }  # => true
+ *    {foo: 0, bar: 1, baz: 2}.one? {|key, value| value < 2  } # => false
```
Comment (JuPlutonic) [inline enum.c:1943]: extra space in block `{|key, value| value < 2 }`

File: enum.c [hunk 3]
```diff
@@ -1950,25 +1969,43 @@ DEFINE_ENUMFUNCS(none)
...
+ *    {foo: 0}.none?(Array)                 # => false
+ *    [].none?(Integer)                     # => true
+ *
+ *  With a block given, returns whether the block returns a truthy value
+ *  for no element:
+ *
+ *    (1..4).none? {|element| element < 1 }                     # => true
+ *    (1..4).none? {|element| element < 2 }                     # => false
+ *    {foo: 0, bar: 1, baz: 2}.none? {|key, value| value < 0 }  # => true
+ *    {foo: 0, bar: 1, baz: 2}.none? {|key, value| value < 1  } # => false
```
Comment (JuPlutonic) [inline enum.c:2005]: extra space

File: enum.c [hunk 4]
```diff
@@ -1950,25 +1969,43 @@ DEFINE_ENUMFUNCS(none)
...
+ *
+ *  Related: #.one?, #all?, #any.
```
Comment (JuPlutonic) [inline enum.c:2007]: correction:
`#one?`, not `#.one?`
`#any?`, not `#any`

File: enum.c [hunk 5]
```diff
@@ -2026,26 +2063,52 @@ min_ii(RB_BLOCK_CALL_FUNC_ARGLIST(i, args))
...
+ *
+ *  - A negative integer if <tt>a < b</tt>.
+ *  - Zero if <tt>a == b</tt>.
+ *  - A positive integer if <tt>a > b</tt>.
+ *
+ *  With a block given and no argument,
+ *  returns the minimum value as determined by the block:
+ *
+ *    %w[xxx x xxxx xx].min {|a, b| a.size <=> b.size } # => "x"
+ *    [].min {|a, b| a <=> b }                          # => nil
```
Comment (JuPlutonic) [inline enum.c:2103]: suggestion:
{foo: 0, bar: 1, baz: 2}.min {|pair1, pair2| pair1[1] <=> pair2[1] }
=> [[:foo, 0], [:bar, 1]]

File: enum.c [hunk 6]
```diff
@@ -2026,26 +2063,52 @@ min_ii(RB_BLOCK_CALL_FUNC_ARGLIST(i, args))
...
+ *
+ *  With a block given and positive integer argument +n+ given,
+ *  returns an array containing the first +n+ minimum elements that exist,
+ *  as determined by the block.
+ *
+ *    %w[xxx x xxxx xx].min(2) {|a, b| a.size <=> b.size } # => ["x", "xx"]
+ *    [].min(2) {|a, b| a <=> b }                          # => []
```
Comment (JuPlutonic) [inline enum.c:2110]: {foo: 0, bar: 1, baz: 2}.min(2) {|pair1, pair2| pair1[1] <=> pair2[1] }
=> [[:foo, 0], [:bar, 1]]

File: enum.c [hunk 7]
```diff
@@ -2026,26 +2063,52 @@ min_ii(RB_BLOCK_CALL_FUNC_ARGLIST(i, args))
...
+ *  returns the minimum value as determined by the block:
+ *
+ *    %w[xxx x xxxx xx].min {|a, b| a.size <=> b.size } # => "x"
+ *    [].min {|a, b| a <=> b }                          # => nil
+ *
+ *  With a block given and positive integer argument +n+ given,
+ *  returns an array containing the first +n+ minimum elements that exist,
+ *  as determined by the block.
+ *
+ *    %w[xxx x xxxx xx].min(2) {|a, b| a.size <=> b.size } # => ["x", "xx"]
```
Comment (JuPlutonic) [inline enum.c:2109]: suggestion:
```ruby
{foo: 0, bar: 1, baz: 2}.min(2) {|pair1, pair2| pair1[1] <=> pair2[1] }
# => [[:foo, 0], [:bar, 1]]
```

File: enum.c [hunk 8]
```diff
@@ -2026,26 +2063,52 @@ min_ii(RB_BLOCK_CALL_FUNC_ARGLIST(i, args))
...
+ *  The block is called with two elements +a+ and +b+, and must return:
+ *
+ *  - A negative integer if <tt>a < b</tt>.
+ *  - Zero if <tt>a == b</tt>.
+ *  - A positive integer if <tt>a > b</tt>.
+ *
+ *  With a block given and no argument,
+ *  returns the minimum value as determined by the block:
+ *
+ *    %w[xxx x xxxx xx].min {|a, b| a.size <=> b.size } # => "x"
```
Comment (JuPlutonic) [inline enum.c:2102]: suggestion:
```ruby
{foo: 0, bar: 1, baz: 2}.min {|pair1, pair2| pair1[1] <=> pair2[1] }
# => [:foo, 0]
```

**Pull request merged**